### DISCUSSION TITLE
Windows, windowed mode and borderless window

### DISCUSSION THREAD

**Discussion by Novicek** [ihhub/fheroes2#5015] [Ideas]
Hey,

been following this project for some months now and am always excited to see how it evolves. Currently though, there are still 2 big issues for me which prevent me from playing this over the original windows version with Verok's patch.

Firstly, I don't understand why there has to be a black console window running alongside the game. If there's a technical reason for it, at least make it hidden.

Secondly, the current fullscreen mode appears to be exclusive fullscreen in Windows and using this mode when the game is running at lower than native resolution ultimately results in 2 issues. Slow alt-tabbing and messing up window sizes and scaling for other application upon exiting the game or alt-tabbing.

Currently, the best option I have with the game is running it in bordered window mode. However, the game does not remember size or position of the window. It also ignores Windows' mouse cursor size setting and on my 4K monitor, it becomes very tiny (fullscreen mode upscales it though).

Lastly, having a borderless windowed mode would be the most preferable way for me. I can currently achieve it with using programs like Borderless gaming, but it would be nice to have it as a native setting.

**Comment by oleg-derevenetz** [comment]
Hi @ihhub I suppose all these issues with exclusive fullscreen mode on Windows may be solved by using the `SDL_WINDOW_FULLSCREEN_DESKTOP` mode. This mode is used on Linux and macOS, but not on Windows - on Windows the `SDL_WINDOW_FULLSCREEN` mode is used. Is there any reason for this difference? Are there some issues with `SDL_WINDOW_FULLSCREEN_DESKTOP` on Windows?
